src/drt/src/pa/FlexPA_prep.cpp

```diff
@@ -2633,6 +2633,37 @@ int FlexPA::getEdgeCost(
   return edge_cost;
 }
 
+std::vector<int> FlexPA::extractAccessPatternFromNodes(
+    const std::vector<FlexDPNode>& nodes,
+    const std::vector<std::pair<frMPin*, frInstTerm*>>& pins,
+    std::set<std::pair<int, int>>& used_access_points,
+    const int max_access_point_size)
+{
+  std::vector<int> access_pattern(pins.size(), -1);
+
+  const int source_node_idx = getFlatIdx(-1, 0, max_access_point_size);
+  const int sink_node_idx = getFlatIdx(pins.size(), 0, max_access_point_size);
+  const FlexDPNode& sink_node = nodes[sink_node_idx];
+  int curr_node_idx = sink_node.getPrevNodeIdx();
+
+  while (curr_node_idx != source_node_idx) {
+    if (curr_node_idx == -1) {
+      logger_->error(DRT, 90, "Valid access pattern not found.");
+    }
```

Comment:
Is this equivalent to counting the number of pins visited?

Reply:
curr_node_idx == -1 means we reached a DP node that does not have an ancestor. For a valid access patterns this should only happen at the source node. If this happens that means that the access pattern has not been solved for this unique instance. I don't see the how it could correlate to the number of pins visited. If you can reproduce the issue I would be glad to address it directly